src/ansys/units/quantity.py

```diff
@@ -61,40 +68,15 @@ def __new__(
                 units = copy_from.units
                 value = copy_from.value
 
-        _value = float(value)
-
-        if quantity_map:
-            units = ansunits.QuantityMap(quantity_map).units
-
-        if dimensions:
-            units = ansunits.Unit(dimensions=dimensions)
-
-        if not isinstance(units, ansunits.Unit):
-            units = ansunits.Unit(units)
-
-        _si_value = (_value + units.si_offset) * units.si_scaling_factor
-
-        return float.__new__(cls, _si_value)
-
-    def __init__(
-        self,
-        value: Union[int, float] = None,
-        units: Union[ansunits.Unit, str] = None,
-        quantity_map: dict = None,
-        dimensions: ansunits.Dimensions = None,
-        copy_from: ansunits.Quantity = None,
-    ):
-        if value == None and not copy_from:
-            raise QuantityError.MISSING_REQUIREMENT()
-
-        if copy_from:
-            if value:
-                units = copy_from.units
+        if _array and not isinstance(value, (float, int)):
+            if isinstance(value, np.array):
+                self._value = value
             else:
-                units = copy_from.units
-                value = copy_from.value
-
-        self._value = float(value)
+                self._value = np.array(value)
+        elif not _array and not isinstance(value, (float, int)):
+            raise QuantityError.REQUIRES_NUMPY()
+        else:
+            self._value = float(value)
```

Comment:
@ansaboutin The above logic could be easier to follow if it simply starts by checking if the value is numerical, and not mention that again.